### ISSUE TITLE
🐛 BUG: Svelte css styles are not picked up in astro@next - regression!

### ISSUE THREAD

**Issue by delight** [snowpackjs/astro#1772]
### What package manager are you using?

pnpm

### What operating system are you using?

Linux - KDE Neon

### Describe the Bug

1. The style of Svelte components doesn't get rendered in astro@next unless the component is hydrated
2. Even full hydrated astro@next seems to reference a separate SvelteCompononent.svelte?svetle&type=style&lang.css (or scss etc.) with a full path /home/user/project/ which ultimately doesn't work

In v0.20.12 the style gets rendered out correctly regardless if hydrated or not + all styled gets impeccably concatenated into a single css at build time (astro v0.20.12). 

I created a little demo project. Simple starter that colors the digit of the SvelteCounter red. Branch `main` is v0.20.12, branch `next` is astro@next 

[Demo Project](https://github.com/delight/demo-css-prob-svelte-in-next)

```
"devDependencies": {
    "@astrojs/renderer-svelte": "^0.2.0-next.0",
    "astro": "^0.21.0-next.1"
  }
```

### Steps to Reproduce

1. `pnpm init astro` using template: "Starter Kit (Generic)"  +  Selecting Svelte as framework.
2. adding some style to the SvelteCounter component
3. so far everything works as expected in v0.20.12 - regardless if dev or build mode.

**Switching to astro@next**  (I usually use pnpm)

1. pnpm i -D astro@next
2. pnpm i -D @astrojs/renderer-svelte@next
3. start astro dev or astro and expect styles

Results: 
- Styles are missing ( wrong ref in head )
- Styles are not concatenated in a single file as it should be and is in (v0.20)
- Styles of the component are not shown

_With hydration some styles are picked up sometimes (not reliable)._

### Link to Minimal Reproducible Example (Optional)

https://github.com/delight/demo-css-prob-svelte-in-next

**Comment by natemoo-re** [comment]
Interesting, thanks for opening an issue! We'll tackle this before we release v0.21.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
